### DISCUSSION TITLE
lazyvim: noice.nvim throwing errors

### DISCUSSION THREAD

**Discussion by sainih09** [LazyVim/LazyVim#1559] [Q&A] [answered]
I'm relatively new to neovim. Everytime I try add some plugins to my config, weird popups from noice.nvim start coming up, cmdline stops working giving more popups.  cmdline infact does execute commands but I can't see anything being typed in the popup box other than warnings.

Running neovim v0.9.2 on MacOS Version 14.0 (similar behaviour was on 13 too).

What am I doing wrong here.

logs from noice.log

```shell
➜  cat noice.log
File: noice.log


Sat Sep 30 22:13:54 2023
....local/share/nvim/lazy/noice.nvim/lua/noice/util/ffi.lua:38: dlsym(RTLD_DEFAULT, syn_attr2entry): symbol not found

stack traceback:
    [C]: in function 'pcall'
    ...local/share/nvim/lazy/noice.nvim/lua/noice/util/call.lua:144: in function <...local/share/nvim/lazy/noice.nvim/lua/noice/util/call.lua:143>
    [C]: in function '__index'
    ....local/share/nvim/lazy/noice.nvim/lua/noice/util/ffi.lua:38: in function '__index'
    ...local/share/nvim/lazy/noice.nvim/lua/noice/util/lazy.lua:16: in function '__index'
    .../share/nvim/lazy/noice.nvim/lua/noice/text/highlight.lua:15: in function 'setup'
    .../share/nvim/lazy/noice.nvim/lua/noice/text/highlight.lua:30: in function 'attr2entry'
    ...
    [C]: in function 'xpcall'
    ...local/share/nvim/lazy/noice.nvim/lua/noice/util/call.lua:149: in function 'try'
    ...local/share/nvim/lazy/noice.nvim/lua/noice/view/init.lua:140: in function 'display'
    .../share/nvim/lazy/noice.nvim/lua/noice/message/router.lua:215: in function <.../share/nvim/lazy/noice.nvim/lua/noice/message/router.lua:155>
    [C]: in function 'xpcall'
    ...local/share/nvim/lazy/noice.nvim/lua/noice/util/call.lua:149: in function <...local/share/nvim/lazy/noice.nvim/lua/noice/util/call.lua:134>
    [C]: in function 'pcall'
    ...local/share/nvim/lazy/noice.nvim/lua/noice/util/init.lua:120: in function ''
    vim/_editor.lua: in function ''
    vim/_editor.lua: in function <vim/_editor.lua:0>
```

**Comment by folke** [comment, marked as answer]
You're running a Neovim version that doesn't seem to be built with LuaJIT or ffi.

**Reply by sainih09** [reply]
Thanks @folke .
I had installed it with brew on M1. installing directly from the neovim releases seem to have solved it.

**Reply by bmulholland** [reply]
I'm hitting this too, and I'm not the only one: https://github.com/folke/noice.nvim/issues/608

Is the homebrew installation of neovim not supported?

**Reply by bmulholland** [reply]
No, something else seems to be up?

```
NVIM v0.9.2
Build type: Release
LuaJIT 2.1.1695653777
```

It's got LuaJIT. How would I know if ffi is included?

**Reply by folke** [reply]
Something must have changed in the homebrew build related to ffi. That error you're getting is about an ffi function is 100% certain in all Neovim versions > 0.5

**Reply by folke** [reply]
Closed that issue. See comments: https://github.com/folke/noice.nvim/issues/608

**Reply by bmulholland** [reply]
Thanks, that explains why installing nightly fixes it. Appreciate the investigation, helps me know I can do something temporary (use nightly), before going back to stable versions when the backported fix is released.

**Comment by folke** [comment]
See https://github.com/folke/noice.nvim/issues/608
